File: app/src/main/java/com/karla/learningverbs/viewmodel/GetPasswordViewModel.java
package com.karla.learningverbs.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.karla.learningverbs.R;
import com.karla.learningverbs.repository.userrepository.UserRepository2;
import com.karla.learningverbs.utils.LearningApplication;
import com.karla.learningverbs.utils.firebase.CustomOnCompleteListener;
import com.karla.learningverbs.kotlin.utils.base.BaseViewModel;

public class GetPasswordViewModel extends BaseViewModel {
    private MutableLiveData<String> isSendEmail;
    private UserRepository2 repository;

    public GetPasswordViewModel() {
        this.repository = UserRepository2.getInstance();
        repository = UserRepository2.getInstance();
        this.isSendEmail = new MutableLiveData<>();
    }

    public void getRecoveryPassword(String email) {

        repository.recoverPassword(email, new CustomOnCompleteListener<Void>() {
            @Override
            public void onSuccess(Void task) {
                isSendEmail.postValue(LearningApplication.getInstance().getString(R.string.msg_email_send));
            }

            @Override
            public void onFailure(Throwable throwable) {
                msgError.postValue(LearningApplication.getInstance().getString(R.string.msg_no_email_send));
            }

            @Override
            public void showLoaging() {
                loading.postValue(true);
            }

            @Override
            public void hideLoading() {
                loading.postValue(false);
            }
        });

    }

    public LiveData<String> sendEmail() {
        return isSendEmail;
    }

}

File: app/src/main/java/com/karla/learningverbs/kotlin/viewmodel/GetPasswordViewModel.kt
package com.karla.learningverbs.kotlin.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.karla.learningverbs.R
import com.karla.learningverbs.kotlin.repository.UserRepository
import com.karla.learningverbs.kotlin.utils.base.BaseViewModel
import com.karla.learningverbs.kotlin.utils.firebase.CustomOnCompleteListener
import com.karla.learningverbs.utils.LearningApplication

class GetPasswordViewModel: BaseViewModel() {

    private val isSendEmail = MutableLiveData<String>()
    private val repository: UserRepository = UserRepository()


    fun getRecoveryPassword(email: String?) {
        repository.recoverPassword(email, object : CustomOnCompleteListener<Void?>() {

            override fun onSuccess(task: Void?) {
                isSendEmail.postValue(
                    LearningApplication.getInstance().getString(R.string.msg_email_send)
                )            }

            override fun onFailure(throwable: Throwable?) {
                msgError.postValue(
                    LearningApplication.getInstance().getString(R.string.msg_no_email_send)
                )
            }

            override fun showLoaging() {
                loading.postValue(true)
            }

            override fun hideLoading() {
                loading.postValue(false)
            }
        })
    }

    fun sendEmail(): LiveData<String?> {
        return isSendEmail
    }
}